src/RobotsHeaders.php

```diff
@@ -41,12 +41,50 @@ public function mayFollow(string $userAgent = '*'): bool
 
     public function noindex(string $userAgent = '*'): bool
     {
-        return $this->robotHeadersProperties[$userAgent]['noindex'] ?? false;
+        $userAgent = strtolower($userAgent);
+
+        if (isset($this->robotHeadersProperties[$userAgent])
+            && isset($this->robotHeadersProperties[$userAgent]['noindex'])) {
+            return $this->robotHeadersProperties[$userAgent]['noindex'];
+        }
+
+        if ($wildCard = $this->checkForWildcard($userAgent, 'noindex') !== null) {
+            return $wildCard;
+        }
+
+        return $this->robotHeadersProperties['*']['noindex'] ?? false;
     }
 
     public function nofollow(string $userAgent = '*'): bool
     {
-        return $this->robotHeadersProperties[$userAgent]['nofollow'] ?? false;
+        $userAgent = strtolower($userAgent);
+
+        // check for exact
```

Comment:
Whenever you use comments like this, ask yourself if you can improve the code in a way the the comment isn't needed. Maybe you can extract such a block to it's own function with the name of the comment.

Reply:
Comment removed and code changed (for the null coalescent).